src/iota/buffer.rs

```diff
@@ -211,6 +247,40 @@ impl Buffer {
 
 }
 
+fn is_alphanumeric_or_(c: char) -> bool {
+    c.is_alphanumeric() || c == '_'
+}
+
+impl WordEdgeMatch {
+    /// If c1 -> c2 is the start of a word.
+    /// If end of word matching is wanted then pass the chars in reversed.
+    fn is_word_edge(&self, c1: &u8, c2: &u8) -> bool {
+        match (self, char::from_u32(*c1 as u32).unwrap(), char::from_u32(*c2 as u32).unwrap()) {
```

Comment:
Both ways have issues with multi-byte characters http://is.gd/M5E7bk. I'm assuming the `u8`s are from the UTF-8 encoding of the text? Gonna have to think of an alternative for this.